webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/precomputed/PrecomputedArray.scala

```diff
@@ -0,0 +1,76 @@
+package com.scalableminds.webknossos.datastore.datareaders.precomputed
+
+import com.scalableminds.webknossos.datastore.datareaders.{
+  AxisOrder,
+  ChunkReader,
+  DatasetArray,
+  DatasetPath,
+  FileSystemStore
+}
+import com.typesafe.scalalogging.LazyLogging
+import play.api.libs.json.{JsError, JsSuccess, Json}
+
+import java.io.IOException
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+
+object PrecomputedArray extends LazyLogging {
+  @throws[IOException]
+  def open(magPath: Path, axisOrderOpt: Option[AxisOrder], channelIndex: Option[Int]): PrecomputedArray = {
+
+    val store = new FileSystemStore(magPath.getParent)
+    val headerPath = s"${PrecomputedHeader.METADATA_PATH}"
+    val headerBytes = store.readBytes(headerPath)
+    if (headerBytes.isEmpty)
+      throw new IOException(
+        "'" + PrecomputedHeader.METADATA_PATH + "' expected but is not readable or missing in store.")
+    val headerString = new String(headerBytes.get, StandardCharsets.UTF_8)
+    val rootHeader: PrecomputedHeader =
+      Json.parse(headerString).validate[PrecomputedHeader] match {
+        case JsSuccess(parsedHeader, _) =>
+          parsedHeader
+        case errors: JsError =>
+          throw new Exception("Validating json as precomputed metadata failed: " + JsError.toJson(errors).toString())
+      }
+
+    val key = magPath.getFileName
+
+    val scaleHeader: PrecomputedScaleHeader = PrecomputedScaleHeader(
+      rootHeader.getScale(key.toString).getOrElse(throw new IllegalArgumentException()),
```

Comment:
maybe add some text to the exception so that if this ever happens we know where to look?